### ISSUE TITLE
Encrypted Files Are Stored (Temporary) in Plaintext And Can Be Downloaded Without Authorization

### ISSUE THREAD

**Issue by jannero** [nilsteampassnet/TeamPass#2192]
### Steps to reproduce
1. Upload JPG image (or probably any file that TeamPass previews when the file link is clicked)
2. Open the file into preview dialog

### Expected behaviour
The files is not stored unencrypted on the file system and is not readable without authentication and authorization.

### Actual behaviour
The files is unencrypted on the disk and you can download it.

For example if the encrypted filename is 123456abcd which is in the upload directory there is 123456abcd_delete.jpg (file extension depends on the file type) on the disk and you can download it without any authorization.

### Server configuration
**Operating system**: Linux

**Web server:** Apache

**Database:** MariaDB

**PHP version:** 7.2

**Teampass version:** 2.1.27.11 

**Teampass configuration file:** not relevant

**Updated from an older Teampass or fresh install:** fresh install

### Client configuration
**Browser:** Firefox, chromium, wget

**Operating system:** Linux

#### Web server error log
```
# Preview is opened
teampass_1  | 10.138.0.34 - - [06/Apr/2018:11:41:52 +0000] "POST /sources/items.queries.php HTTP/1.0" 200 630 "https://teampass-staging.intra.vincit.fi/index.php?page=items" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.181 Safari/537.36"

# Downloaded the file with wget (no authorization at all) and you get the unencrypted version of the file
teampass_1  | 10.138.0.34 - - [06/Apr/2018:11:41:58 +0000] "GET /upload/53666b138b04358bb418c632c716cd32_delete.jpg HTTP/1.0" 200 54961 "-" "Wget/1.19.4 (linux-gnu)"

# Preview view is closed (this request seems to remove the file from the disk)
teampass_1  | 10.138.0.34 - - [06/Apr/2018:11:42:02 +0000] "POST /sources/main.queries.php HTTP/1.0" 200 291 "https://teampass-staging.intra.vincit.fi/index.php?page=items" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.181 Safari/537.36"
```

**Comment by nilsteampassnet** [comment]
You are right.

This is due to the fact that the `files` folder is in the "site domain".
You can restrict access by using .htaccess rules.

But I've improved this.
I did a change of the code to permit to have the "files" folder outside the teampass site.
This permits you to define a folder anywhere on your server (but requires Apache user to have all rights on it) so that any one can access the temporary clear file.
And I also added a special option that doesn't require a temporary clear file. If enabled, the image content is exchanged from server to client. This is more secure but needs more time to show the image.

I believe those 2 changes improve highly Teampass's security ... thank you for the suggestion.

Commit is done in development branch and will be part of 2.1.27.12.

Reopen on need.

**Comment by jannero** [comment]
Thanks for the quick change. I tested the latest version in _master_ and now there is an error:

`Error. The file is missing`

The HTTP response body is:

`Unknown column 'content' in 'field list'TfrKWnJycnJOtBipS7H+lKWbw+CVL/Mso4qnb2sBCCM9HUk0NcQ8wdGGytaz77prrVDE9Sagv6L2Tj9tN8xROevbAARlhIilpsi26DJ3mmv+wyuH/+Eskyl/oYdGVZt08zQ27hoYT/Mrcg==`

Also as you can see from the web server error log the files are kept in the upload directory (the files directory is empty all the time).

The same error happens for the file (PNG image) I had uploaded previously as with newly uploaded file.

**Comment by nilsteampassnet** [comment]
The change was not commiting master only in development.
Run upgrade after overwriting the existing files.

**Comment by jannero** [comment]
Ah true, I actually used latest commit in development branch (commit 6d75404a2d2d15aaebc8fff210ce0741366ddc6b)

**Comment by jannero** [comment]
I tested the case described in https://github.com/nilsteampassnet/TeamPass/issues/2189 and it was fixed (and I can see that the same commit contains that fix as the change that should fix this issue)

**Comment by jannero** [comment]
@nilsteampassnet Did you miss the above comment?

**Comment by nilsteampassnet** [comment]
@jannero 
Yes I did but I'm pretty lost.
What is fixed or not for you? as the 2 tickets have nothing in common ;)

**Comment by jannero** [comment]
@nilsteampassnet I referred the other issue to make clear that I tested commit that contained fixes for both issues.

So the 2FA does work as per my testing but now the previewed images are not displayed.

So to produce the current issue (I'm testing with commit 6d75404a2d2d15aaebc8fff210ce0741366ddc6b):

- Upload image (I used PNG image)
- Click the image link

Now the Information  dialog is opened and it is empty. An error message "Error: The file is missing" is shown.

In Apache logs you see:

`[13/Apr/2018:05:04:55 +0000] "POST /sources/items.queries.php HTTP/1.0" 200 529 "https://teampass-staging.intra.vincit.fi/index.php?page=items" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.181 Safari/537.36"`

The body of that response is:

```
Unknown column 'content' in 'field list'dzrQWp+fn5/YLTOwVtfDPqboncOMeHsCIUHXvAi+5IP5HijfIk0WSL134Ioa+ji2LfjUxAe2YjRLM8yhbkjuShuyreFaNwlYK4h37FnhMItGvcTdKxP6WuezIb69HL6brzMMhgGjm+2x7w==
```

When the dialog is open, there is no _delete file anymore.

But note that the uploaded file is still saved in upload directory. I would guess that the fix that was made changed the code in such way that it tries to read the file from the files directory (this is just an assumption/guess based on the comments you made).
